src/EditColumnsModal/EditColumnsModal.tsx

```diff
@@ -0,0 +1,104 @@
+import { LfLocalizationService } from "@laserfiche/lf-js-utils";
+import React from "react";
+import { ColumnDef } from '@laserfiche/types-lf-ui-components';
+import './EditColumnsModal.css';
+
+const resources: Map<string, object> = new Map<string, object>([
+    ['en-US', {
+      'NAME': 'Name',
+      'OK': 'Ok',
+      'CANCEL': 'Cancel',
+      'ADD_REMOVE_COLUMNS': 'Add/Remove Columns',
+    }],
+    ['es-MX', {
+      'NAME': 'Name -Spanish',
+      'OK': 'Ok - Spanish',
+      'CANCEL': 'Cancel - Spanish',
+      'ADD_REMOVE_COLUMNS': 'Add/Remove Columns - Spanish',
+    }]
+]);
+
+export default class EditColumnsModal extends React.Component<{onClose: (columns?: ColumnDef[]) => Promise<void>; errorMessage: string; initialColumnsSelected: ColumnDef[];allColumnOptions: ColumnDef[] }, {columnsSelected: ColumnDef[]; open: boolean}> {

```

Comment:
since we are creating these new, I would create them as function components, we can mob on that